### ISSUE TITLE
Slow init on iPhone 5s

### ISSUE THREAD

**Issue by galambalazs** [life360/confetti#10]
Great project. It has some problems though. When running on a 5s bursting 100 confettis the start lags sometimes. Takes all the fun away.
(not sure if Pedometer++ uses this code or something similar but I noticed the initial lag there too).

I was looking at your code: 

    - (void)burstAt:(CGPoint)point confettiWidth:(CGFloat)confettiWidth 
      numberOfConfetti:(NSInteger)numberConfetti

Timing it only 18 ms is spent on the background thread vs 137 ms on the main thread. Seems like all that dispatch dance should be removed.

(also I don't think you need to use `weakSelf` since you are not storing the blocks on `self` so there is no circular reference going back and forth between them).

My proposed solution:

- Do *all* the work on the main thread (switching between main & bg has a cost which doesn't worth it here as I showed above)
- Batch every 5 confetti inits to a main thread `block` so you still let it breath

**Comment by galambalazs** [comment]
The only way I can get acceptable performance right now is if the confettis don't start all at once but I use dispatch_after and space it out a bit.

**Comment by mmislam101** [comment]
@galambalazs Thanks for the good suggestions. I'll try and make some time to batch them and dispatch on the main thread and see if it solves the latency issues. Thanks a lot for also doing the work to profile the timing, it's useful information. 

We're also totally open to anyone contributing a Pull Request with the changes and I'll review it right away and merge it if it looks good and update the podspec accordingly.

**Comment by galambalazs** [comment]
Sure if I have an improved version I'll do a PR.

**Comment by mmislam101** [comment]
@galambalazs yes please share any improved versions with us. We'd welcome it.
